### ISSUE TITLE
Cache log signature

### ISSUE THREAD

**Issue by UrknallBigBang** [2Abendsegler/GClh#2424]
### Describe the bug

No longer shows cache log signature in the log text and the tb's are no longer marked with auto visited.

### Expected behavior

nothing happens no text no tb mark

### OS

Windows

### Browser

Firefox

### GClh Version

0.14.6

**Comment by 2Abendsegler** [comment]
Thanks for reporting the bug.

The reason should be related to the [revision of the entire page](https://forums.geocaching.com/GC/index.php?/topic/394827-release-notes-website-new-geocache-and-trackable-logging-flow-november-2-2023/).
The changes are currently being rolled out. I don't see the changes myself at the moment. As soon as the changes are visible to us, we will start making the necessary adjustments.

**Comment by UrknallBigBang** [comment]
Thx for the fast reply. Danke für die schnelle Antwort. Und danke für das tolle Userscript, kann man euch irgendwie unterstützen?

**Comment by 2Abendsegler** [comment]
Wenn du uns unterstützen möchtest, dann gibt es hier bei GitHub auf der [GClh Hauptseite](https://github.com/2Abendsegler/GClh) einen Sponsor Button. 
![Unbenannt](https://github.com/2Abendsegler/GClh/assets/22332216/33410793-fe81-4f1d-9f38-dd1862ab73cf)
Dort aufgelistet ist auch ein Weg über Paypal, bei dem keine Gebühren abgezogen werden, nur falls du auch selbst einen Paypal Account hast.

**Comment by capoaira** [comment]
Oh no, das sieht nach einer Menge Arbeit aus... da könnten wir sicherlich auch Unterstützung gebrauchen 😅

Aber eigentlich sehen die Änderungen gut aus. Schade, dass ich sie noch nicht sehen kann...

**Comment by ztNFny** [comment]
> Schade, dass ich sie noch nicht sehen kann...

Anyone can access the new log pages, you just have to know the URL: https://www.geocaching.com/live/geocache/GC40/log

**Comment by capoaira** [comment]
> > Schade, dass ich sie noch nicht sehen kann...
> 
> Anyone can access the new log pages, you just have to know the URL: https://www.geocaching.com/live/geocache/GC40/log https://www.geocaching.com/live/trackable/TBxxxx/log

Thanks. That helps. I will take a look and will create a to-do list for the new-new logging

**Comment by capoaira** [comment]
Cache and TB logging
- [ ] Log Templates
- [ ] Replace log by last log template
- [ ] Signature
- [ ] Add log length (characters and words). New limitation: 5000 (before 4000)
- [ ] Show message in case of unsaved log
- [ ] Show additional cache info
- [ ] Show Pseudonym (part of additional cache info???)
- [ ] Include VIP/VUP/message/mail for owner (Link uses the `id`, not `guid` and not the PR code)
- [ ] Default log type
- [ ] Default event log type
- [ ] Default owner log type
- [ ] Maybe a compact layout?

Only Cache Logging
- [ ] Auto Visit
  - [ ] Set "AutoVisit" for all TBs by default
- [ ] Default TB log type

Not know behavior:
- After sending a draft related log, automatic go back to drafts
- After sending a draft related log, automatic view log
- After sending a non draft related log, automatic view log

I'm not exactly know what `Replace placeholder owner, although there could be a pseudonym` means. But the new-new Logform shows the real name and not the pseudonym.

What we did no longer need:
- Log Preview
- Show Smiles

@2Abendsegler have I forgot some features?

**Comment by 2Abendsegler** [comment]
@capoaira 
Sehr gute und ausführliche Liste, super!

Bei den Parametern "After sending a draft related log ..." and "After sending a non draft related log, automatic view log" geht es um das Verhalten, nachdem das Log erzeugt wurde und wir auf die Cache Seite weitergeleitet wurden. Ausschlaggebend sind die Informationen, die dann auf der Cache Seite angezeigt werden. Die Parameter sollen dann für ein Weiterleiten zu den Drafts bzw. zur Anzeige des Logs sorgen. 
Die Parameter sollten wir alle kurz testen, kann ich übernehmen.
Also vielleicht einen weiteren Punkt aufnehmen: `Automatic after sending a draft related or non draft related log`

Als Owner wird nicht überall der Owner des Caches angezeigt, sondern das Pseudonym das der Owner im Cache angegeben hat. Beispielsweise kann man das Pseudonym oder den Owner auch wahlweise im Cache Listing oben bei "a cache by" anzeigen lassen. So weit ich mich erinnere wurde im neuen Log Formular bisher eben dieses Pseudonym angezeigt, falls es vorhanden ist. Der Parameter steuert nun ob man auch damit einverstanden ist, dieses Pseudonym als Platzhalter für den Owner zu verwenden.
Wir sollten kurz testen ob immer noch ein vorhandenes Pseudonym hier angezeigt wird und ob der Parameter dann noch funktioniert. Das kann ich übernehmen.
Also vielleicht einen weiteren Punkt aufnehmen: `Replace owner pseudonym`

Wir sollten noch testen, ob das alte Log Formular aus den Drafts bzw. als Log noch wie gewohnt erreichbar ist. Das kann ich übernehmen.
Ein weiterer Punkt also: `Logging and draft logging with old log form`
 
Wir sollten zumindest grob das Ganze auch als `Basis Member` testen.

Bei den zu entfernenden Themen "Log Preview" und "Show Smiles" sollten wir auch das alte Log Formluar gegenchecken. 

Vielleicht können wir gemeinsam dran arbeiten? Wie wollen wir die Punkte auswählen? Sollen wir uns mit Namen in deiner Liste eintragen oder sollen wir einfach einen Eintrag per Checkbox markieren, an dem wir arbeiten? Falls man mit einem Eintrag nicht klar kommt, kann man die Markierung ja auch wieder entfernen und dann ist der Eintrag wieder frei zur Bearbeitung.

**Comment by ztNFny** [comment]
> Wir sollten noch testen, ob das alte Log Formular aus den Drafts bzw. als Log noch wie gewohnt erreichbar ist. Das kann ich übernehmen.

Users switched to the new workflow can't use the old URLs anymore, they redirect to the new ones.

**Comment by 2Abendsegler** [comment]
> Users switched to the new workflow can't use the old URLs anymore, they redirect to the new ones.

Vielleicht kannst du mal [das hier](https://forums.geocaching.com/GC/index.php?/topic/343005-gc-little-helper-ii/&do=findComment&comment=6106197) versuchen.

**Comment by ztNFny** [comment]
No, even the direct calls to the sites don't work anymore, we checked that for c:geos logging workflow. The old sites will be removed completely "soon".

**Comment by capoaira** [comment]
> Vielleicht können wir gemeinsam dran arbeiten? Wie wollen wir die Punkte auswählen? Sollen wir uns mit Namen in deiner Liste eintragen oder sollen wir einfach einen Eintrag per Checkbox markieren, an dem wir arbeiten?

Ja gerne. Allerdings weiß ich nicht, wie viel ich helfen kann. Ich würde tatsächlich zu jedem Feature ein Issue erstellen. Falls es Probleme gibt, kann man am übersichtlichen diskutieren.

Ich würde gerne versuchen hier mehr mit asynchronen Funktionen (`async` / `await`) zu arbeiten, falls die Seite asynchron lädt.

> Users switched to the new workflow can't use the old URLs anymore, they redirect to the new ones.

That would be great for us. Makes things easier. And we can clean up our code.


> Btw: the new log pages have all the interesting stuff as json objects in global scope - very nice for accessing stuff.

Oh wow, thanks. That will help. Sadly The data is not correct like the T/D-Raiting and Favs :-(

**Comment by ztNFny** [comment]
> Ich würde gerne versuchen hier mehr mit asynchronen Funktionen (async / await) zu arbeiten, falls die Seite asynchron lädt

Everything is included in the html in that big json object, no async loading of anything relevant.

**Comment by 2Abendsegler** [comment]
> No, even the direct calls to the sites don't work anymore
Es handelt sich bei dem angegebenen Link in meinem Beitrag im Forum nicht um einen direkten Link zum alten Log Formular sondern um den Optout Link zum alten Log Formular. Die Optout Funktionalität funktioniert noch, auch wenn man sie nicht mehr aus dem neuen Log Formular anwählen kann. Hast du es mal getestet, wie ich es beschrieben habe?

**Comment by capoaira** [comment]
Vielleicht wäre dies der richtige Zeitpunkt, die Placeholder um den Placeholder `#ownerPseudonym#` zu erweitern?

@ztNFny, @UrknallBigBang: What is the link to the new log form, if you come from a draft?

**Comment by 2Abendsegler** [comment]
> Vielleicht wäre dies der richtige Zeitpunkt, die Placeholder um den Placeholder `#ownerPseudonym#` zu erweitern?

Ich glaube es war bisher im neuen Log Formular nicht erkennbar, ob der angezeigte Owner ein echter Owner oder ein Pseudonym ist. Deshalb der Parameter, dass auch gegebenenfalls das Pseudonym als Owner verwendet werden darf. Wird der Parameter nicht aktiviert, wird der Platzhalter für den Owner im neuen Log Formular gar nicht ersetzt.

**Comment by capoaira** [comment]
> > Vielleicht wäre dies der richtige Zeitpunkt, die Placeholder um den Placeholder `#ownerPseudonym#` zu erweitern?
> 
> Ich glaube es war bisher im neuen Log Formular nicht erkennbar, ob der angezeigte Owner ein echter Owner oder ein Pseudonym ist. Deshalb der Parameter, dass auch gegebenenfalls das Pseudonym als Owner verwendet werden darf. Wird der Parameter nicht aktiviert, wird der Platzhalter für den Owner im neuen Log Formular gar nicht ersetzt.

Ah ok. Jetzt ist es immer der Name und nicht das Pseudonym des Owners

**Comment by 2Abendsegler** [comment]
> Ah ok. Jetzt ist es immer der Name und nicht das Pseudonym des Owners

Ja, stimmt. Ich habe mir das auch gerade mal angesehen. Den Parameter und die Logik dazu können wir dann rausschmeißen.

**Comment by capoaira** [comment]
@2Abendsegler Wie wollen wir das ganze angehen?
Für jeden Punkt ein Issue ist vielleicht doch etwas viel. Einige Sachen wie z.B. die Loglänge sind sehr einfach einzubauen. Dennoch wäre es gut, sich die einzelnen Sachen assignen zu können (für die bessere Übersicht)
Ich habe jetzt mal eine grobe Struktur mit Kommentaren geschaffen, damit wir später keine merge Konflikte bekommen. Wenn wir ein Feature einbauen, können wir das unter dem jeweiligen Kommentar machen (PR kommt gleich, dort ist die Loglänge dann auch mit drin)

**Comment by 2Abendsegler** [comment]
Das hier sind genau genommen neue Features für das neue Log Formular. Die gab es nur im alten Log Formular. Die sollten wir hinten anstellen, wie auch andere neue Features.
Default log type
Default event log type
Default owner log type
Default TB log type

**Comment by 2Abendsegler** [comment]
... ist auch neu (machen wir natürlich jetzt fertig).
F2 key, Log abschicken

**Comment by capoaira** [comment]
Na ja, so wirklich neu sind die Default Logtypen ja nicht. Zumindest für Cacher, die die ganze Zeit über das alte Formular genutzt haben. Allerdings habe ich mich gerade daran versucht und es einfach nicht geschafft. Leider ist es kein einfaches `select`, sondern irgendeine fancy Scheiße, die wir schlecht bzw. gar nicht simulieren können :(

**Comment by 2Abendsegler** [comment]
> Na ja, so wirklich neu sind die Default Logtypen ja nicht. Zumindest für Cacher, die die ganze Zeit über das alte Formular genutzt haben. Allerdings habe ich mich gerade daran versucht und es einfach nicht geschafft. Leider ist es kein einfaches `select`, sondern irgendeine fancy Scheiße, die wir schlecht bzw. gar nicht simulieren können :(

Ja, fancy Scheiße, genau. Und ja, im alten Formular gabs die, aber eben nicht im neuen. Ich hatte mich auch schon erfolglos daran versucht. Das Issue hier oder ein neues Issue mit den noch offenen Punkten zum Log Formular werden wir ja behalten. Ich schlage vor wir werden das Thema nach der nächsten Version in Ruhe versuchen.

**Comment by 2Abendsegler** [comment]
@capoaira 
Moin,
wollte ja eigentlich heute eine neue Version raushauen. Schaffe ich aber nicht. Es sind einfach noch zu viele Baustellen. 😓
Als nächsten Termin werde ich nächsten Sonntag ins Auge fassen.

**Comment by 2Abendsegler** [comment]
@capoaira
Aus meiner Sicht sind noch die folgenden Punkte offen. Ich würde aber keinen von den Punkten im Moment realisieren wollen. Soll ich trotzdem dafür Issues aufmachen, oder sollen wir die Punkte vergessen?

- Maybe a compact layout? (new)
- After edit a log, go back to log view (new)

Ich würde das Issue gerne schließen. Möchtest du von hier noch etwas anderes in ein neues Issue übernehmen?

**Comment by capoaira** [comment]
Ich denke "After edit a log, go back to log view" wäre schön. Das kompakte Layout können wir gerne erstmal vergessen. Lass uns einfach ein neues Issue dafür aufmachen.
